File: python/code_examples/04-classes/main.py
class Greeter:
    def __init__(self, name: str) -> None:
        self.name = name

    def hello(self) -> str:
        return f"Hello, {self.name}"


class Admin(Greeter):
    def hello(self) -> str:
        return super().hello() + " (admin)"


def main() -> None:
    g = Admin("Ada")
    print(g.hello())


if __name__ == "__main__":
    main()

File: python/code_examples/04-classes/main.js
class Greeter {
  constructor(name) {
    this.name = name;
  }
  hello() {
    return `Hello, ${this.name}`;
  }
}

class Admin extends Greeter {
  hello() {
    return `${super.hello()} (admin)`;
  }
}

function main() {
  const g = new Admin("Ada");
  console.log(g.hello());
}

main();
